prow/cmd/crier/README.md

```diff
@@ -0,0 +1,60 @@
+# Crier
+
+Crier reports your prowjobs on their status changes.
+
+## Usage / How to enable existing available reporters
+
+### [Gerrit reporter](/prow/gerrit/reporter)
+
+You can enable gerrit reporter in crier by specifying `--gerrit` flag.
+
+Similar to the [gerrit adapter](/prow/cmd/gerrit), you'll need to specify `--gerrit-projects` for
+your gerrit projects, and also `--cookiefile` for the gerrit auth token (leave it unset for anonymous).
+
+
+### [Pubsub reporter](/prow/pubsub/reporter)
+
+You can enable pubsub reporter in crier by specifying `--pubsub` flag.
+
+You need to specify following labels in order for pubsub reporter to report your prowjob:
+
+| Label                          | Description                                                                                                                               |
+| ------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------|
+| `"prow.k8s.io/pubsub-project"` | Your gcp project where pubsub channel lives                                                                                               |
+| `"prow.k8s.io/pubsub-topic"`   | The [topic](https://cloud.google.com/pubsub/docs/publisher) of your pubsub message                                                        |
+| `"prow.k8s.io/pubsub-runID"`   | A user assigned job id. It's tied to the prowjob, serves as a name tag and help user to differentiate results in multiple pubsub messages |
+
+<!-- TODO(krzyzacy): move github reporter over -->
+
+## Implementation details
+
+Crier supports multiple reporters, each reporter will become a crier controller. Controllers
```

Comment:
The gerrit and pubsub reporters are referred to as controllers, clients, and reporters. This would be a little easier to understand with consistent terminology.

Reply:
So there's reporter, and crier controller. A crier controller takes in a reporter interface. Got rid of `clients` now, hope it makes more sense.